### PR TITLE
Fixed multiple graphs crashing

### PR DESCRIPTION
**Pull request by hydra41**
- Shader modules in FPS and Axes were static when they shouldn't be
- Added destroy() to Renderer and explicitly call destroy on renderables as there were validation errors from the renderables being destroyed from finalize (because they were still being held by command buffer memory)
- Deleted some test shaders no longer being used
- Cleaned up some vars in Canvas

### CODE REVIEW

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKRenderer.java [hunk 1]
```diff
@@ -118,6 +118,20 @@ public void AddRenderable(CVKRenderable renderable) {
         }   
     }
     
+    
+    /**
+     * Called from the parent CVKVisualProcessor when the VisualManager is
+     * being destroyed.
+     */
+    public void Destroy() {
+        // TODO HYDRA: Not sure if this is the best method, can it cause a
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKRenderer.java]: Doesn't look like there is.  If you are waiting on named semaphores or fences I think you can set a timeout, but the device only has one wait and it has no timeout value.  So amend your comment?

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKRenderer.java [hunk 2]
```diff
@@ -118,6 +118,20 @@ public void AddRenderable(CVKRenderable renderable) {
...
+        // deadlock? Would it be better to have a shutdown flag and do in the
+        // update loop?
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKRenderer.java]: Not sure we can?  I haven't looked into it but would assume the TopComponent kills us.  It owns the canvas so we can't hang around if it deletes it from under us.  Take a look?

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java [hunk 3]
```diff
@@ -99,7 +99,7 @@ void addTask(final CVKRenderableUpdateTask task) {
     @Override
     protected final void destroyCanvas() {
         //TODO_TT: is this needed?
-        System.out.println("GLVisualProcessor.destroyCanvas called");        
+        System.out.println("CVKVisualProcessor.destroyCanvas called");
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java]: Just delete this print to System.out which we never see anyway.  Not sure what the cleanup needs to be here.  Want to look into it?

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java [hunk 4]
```diff
@@ -232,7 +232,8 @@ protected void initialise() {
     @Override
     protected void cleanup() {
         // TODO_TT:
-        //canvas.removeGLEventListener(renderer);
+        cvkRenderer.Destroy();
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java:234]: Can probably remove the comments above and below.  It would be good if you could dig into the destruction a bit.  See if you can find a way to spot memory leaks as we almost certainly have graph resources we aren't releasing.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java [hunk 5]
```diff
@@ -248,27 +252,48 @@ private static int LoadShaders(CVKDevice cvkDevice) {
...
+    }
+    
+    
+    private int CreateShaderModules() {
+        int ret = VK_SUCCESS;
+        
+        try{           
+            hVertexShaderModule = CVKShaderUtils.createShaderModule(vertexShaderSPIRV.bytecode(), cvkDevice.GetDevice());
+            if (hVertexShaderModule == VK_NULL_HANDLE) {
+                throw new RuntimeException("Failed to create shader from PassThru.vs.spv bytes");
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: This might be a good candidate for returning one of our own error codes?  Can you see if it's possible to fail the creation of a graph without taking Constellation down please?

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java [hunk 6]
```diff
@@ -248,27 +252,48 @@ private static int LoadShaders(CVKDevice cvkDevice) {
...
 
         try{
-            SPIRV vertShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThru.vs", VERTEX_SHADER);
-            SPIRV geomShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThruLine.gs", GEOMETRY_SHADER);
-            SPIRV fragShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThru.fs", FRAGMENT_SHADER);
-            
-            hVertShaderModule = CVKShaderUtils.createShaderModule(vertShaderSPIRV.bytecode(), cvkDevice.GetDevice());
-            hGeomShaderModule = CVKShaderUtils.createShaderModule(geomShaderSPIRV.bytecode(), cvkDevice.GetDevice());
-            hFragShaderModule = CVKShaderUtils.createShaderModule(fragShaderSPIRV.bytecode(), cvkDevice.GetDevice());
+            vertexShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThru.vs", VERTEX_SHADER);
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: I think we need a quick check to see if we need to recompile shaders.  We could do this quickly by checking for <shader>.md5 in the compiled folder.  If it doesn't exist, or doesn't match the md5 we calculate we compile the shader then write the .md5.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java [hunk 7]
```diff
@@ -304,9 +331,16 @@ public static int StaticInitialise(CVKDevice cvkDevice) {
     
     public CVKAxesRenderable(CVKVisualProcessor visualProcessor) {
         parent = visualProcessor;
-      
     }
     
+    
+    public void DestroyStatic() {
+        vertexShaderSPIRV.free();
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: What's calling this?  Freed resources should be nulled so Java can clean up anything it allocated for that resource.  We also need to reset the static initialised flag too.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java (hunk 3 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java]: Deleted old comments, added Destroy() to canvas and removed component listener

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java (hunk 4 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java:234]: Removed old comments

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKRenderer.java (hunk 2 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKRenderer.java]: Removed comment

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 7 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: We don't have a good place to call this atm, ideally it would be when constellation starts and then shutsdown.  For now I'll call when the graph is destroyed and add a comment.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 6 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: I agree will add the task to list to do at a later time

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java [hunk 8]
```diff
@@ -957,6 +991,7 @@ public int DestroySwapChainResources(){
             // different position for our FPS.  After updating the uniform buffers we
             // need to update the descriptor sets that bind the uniform buffers as well.
             DestroyPipeline();
+            DestroyPipelineLayouts();
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: Why did you add this?

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java [hunk 9]
```diff
@@ -1092,14 +1127,26 @@ public boolean NeedsDisplayUpdate() {
     
     @Override
     public int Initialise(CVKDevice cvkDevice) {
+        int ret = VK_SUCCESS;
+        
         this.cvkDevice = cvkDevice;
         
         // This only needs to be initialised once but can't be static as each graph will
         // have their own device and the layout must be bound to that.
         if (hDescriptorLayout == VK_NULL_HANDLE) {
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: Those checks against null made sense when the handle was static, they should be replaced with CVKAssert(h == null) now to make sure we aren't double initialising.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java [hunk 10]
```diff
@@ -325,41 +329,52 @@ private static int LoadShaders(CVKDevice cvkDevice) {
         int ret = VK_SUCCESS;
         
         try {
-            ByteBuffer vsBytes = LoadFileToDirectBuffer(CVKShaderPlaceHolder.class, "compiled/SimpleIcon.vs.spv");
+            vsBytes = LoadFileToDirectBuffer(CVKShaderPlaceHolder.class, "compiled/SimpleIcon.vs.spv");
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: The shader load code should be unified somewhere with the md5 thing I mentioned above.  Good thing you added a shader utils class ;)

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java [hunk 11]
```diff
@@ -1118,8 +1133,14 @@ private void DestroyUniformBuffers() {
         }        
     }
     
+    public void DestroyStatic() {                   
+        MemoryUtil.memFree(vsBytes);
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: See comment on the other DestroyStatic.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java [hunk 12]
```diff
@@ -1274,19 +1295,30 @@ public void IncrementDescriptorTypeRequirements(CVKDescriptorPoolRequirements re
...
     @Override
     public int Initialise(CVKDevice cvkDevice) {
+        int ret = VK_SUCCESS;
+        
         this.cvkDevice = cvkDevice;
         
-        if (hDescriptorLayout == VK_NULL_HANDLE) {
-            int ret = CreateDescriptorLayout(cvkDevice);
-            if (VkFailed(ret)) { return ret; }
+        if (hVertexShader == VK_NULL_HANDLE) {
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: See comment on the last initialise, ie change these to asserts.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 8 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: Because vkCreatePipelineLayout is called in CreatePipeline(), so previously we were re-creating the layouts without destroying them.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 5 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: Added our own return code in the catch block and return it.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java [hunk 13]
```diff
@@ -231,8 +229,9 @@ protected void initialise() {
 
     @Override
     protected void cleanup() {
-        // TODO_TT:
-        //canvas.removeGLEventListener(renderer);
+        cvkAxes.DestroyStaticResources();
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java:232]: But why?  They are static, so destroying them with the instance makes no sense.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 5 above)
Comment (TheTimurid) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: That looks wrong, catching your own throw.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java [hunk 14]
```diff
@@ -244,31 +249,64 @@ private void CopyTo(ByteBuffer buffer) {
...
             
-            hVertShaderModule = CVKShaderUtils.createShaderModule(vertShaderSPIRV.bytecode(), cvkDevice.GetDevice());
-            hGeomShaderModule = CVKShaderUtils.createShaderModule(geomShaderSPIRV.bytecode(), cvkDevice.GetDevice());
-            hFragShaderModule = CVKShaderUtils.createShaderModule(fragShaderSPIRV.bytecode(), cvkDevice.GetDevice());
+            if (geometryShaderSPIRV == null) {
+                geometryShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThruLine.gs", GEOMETRY_SHADER);
+            }
+            
+            if (fragmentShaderSPIRV == null) {
+                fragmentShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThru.fs", FRAGMENT_SHADER);
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java:275]: compileShaderFile should be renamed and changed to match the rest of our methods, return an error code instead of throw.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java [hunk 15]
```diff
@@ -294,8 +332,9 @@ public static int StaticInitialise(CVKDevice cvkDevice) {
         int ret = VK_SUCCESS;
         if (!staticInitialised) {
             LoadShaders(cvkDevice);
-            if (VkFailed(ret)) { return ret; }
-            //ret = CreateDescriptorLayout(cvkDevice);
+            if (VkFailed(ret)) {
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: I think having it on the one line, like the many other instances, is fine.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 8 above)
Comment (TheTimurid) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: Then that needs to change!  Pipeline layouts should be created only once, they are templates, I don't think there is any reason to tie them to the swapchain lifetime.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 14 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java:275]: removed the throw statements as it is in a try/catch anyway.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java (hunk 13 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java:232]: As I said in my previous comment, we don't have a good place to do this yet. Will do in the work with the md5 hash checks.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 8 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: I fixed Axes and FPS so they don't re-create any pipelines by making the viewport/scissor dynamic

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java (hunk 13 above)
Comment (TheTimurid) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java:232]: I don't think we should do any static cleanup.  Just remove it?  Either way, add a task to GitLab please to revisit the static cleanup (very low priority imo).

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKRenderer.java (hunk 1 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKRenderer.java]: Comment removed

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 5 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: Removed throws, just catching the exception as stated below

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 9 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: changed to asserts

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java (hunk 10 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: Added to shader md5 task

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java (hunk 11 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: Added checks/sets for null

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java (hunk 13 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKVisualProcessor.java:232]: Added to shader task

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java [hunk 16]
```diff
@@ -244,31 +251,55 @@ private void CopyTo(ByteBuffer buffer) {
...
-    private static int LoadShaders(CVKDevice cvkDevice) {
+    private static int LoadShaders(CVKDevice cvkDevice) {       
         int ret = VK_SUCCESS;
 
         try{
-            SPIRV vertShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThru.vs", VERTEX_SHADER);
-            SPIRV geomShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThruLine.gs", GEOMETRY_SHADER);
-            SPIRV fragShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThru.fs", FRAGMENT_SHADER);
+            if (vertexShaderSPIRV == null) {
+                vertexShaderSPIRV = compileShaderFile(CVKShaderPlaceHolder.class, "PassThru.vs", VERTEX_SHADER);
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java:259]: There should still be a null check, just return an error code instead.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 15 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: changed to a one liner

Comment (TheTimurid) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java]: This needs to match the other 73 instances.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java [hunk 17]
```diff
@@ -1019,6 +1053,16 @@ private int CreatePipelines() {
                 colorBlending.pAttachments(colorBlendAttachment);
                 colorBlending.blendConstants(stack.floats(0.0f, 0.0f, 0.0f, 0.0f));
 
+                // ===> DYNAMIC PROPERTIES CREATION <===
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java:1078]: Did you test the FPS moves to the right location following a resize?

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 16 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java:259]: It does a null check and returns an error code and logs message in the except block?

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java [hunk 18]
```diff
@@ -410,8 +428,9 @@ public static int StaticInitialise(CVKDevice cvkDevice) {
         int ret = VK_SUCCESS;
         if (!staticInitialised) {
             LoadShaders(cvkDevice);
-            if (VkFailed(ret)) { return ret; }
-//            ret = CreateDescriptorLayout(cvkDevice);
+            if (VkFailed(ret)) {
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: Change this to be consistent please.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java [hunk 19]
```diff
@@ -1274,19 +1328,30 @@ public void IncrementDescriptorTypeRequirements(CVKDescriptorPoolRequirements re
...
+        
+        int ret = VK_SUCCESS;
+        
         this.cvkDevice = cvkDevice;
         
-        if (hDescriptorLayout == VK_NULL_HANDLE) {
-            int ret = CreateDescriptorLayout(cvkDevice);
-            if (VkFailed(ret)) { return ret; }
+        ret = CreateShaderModules();
+        if (VkFailed(ret)) {
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: Change it to be consistent please.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java [hunk 20]
```diff
@@ -1274,19 +1328,30 @@ public void IncrementDescriptorTypeRequirements(CVKDescriptorPoolRequirements re
...
+            return ret;
         }
-        
+         
+        ret = CreateDescriptorLayout(cvkDevice);
+        if (VkFailed(ret)) {
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: Change it to be consistent please.

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java (hunk 17 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java:1078]: yep

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java (hunk 18 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: changed to one liner

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java (hunk 20 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: changed all VkFailed to be one liners

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java (hunk 19 above)
Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java]: changed all VkFailed to be one liners

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java (hunk 16 above)
Comment (TheTimurid) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java:259]: I don't see any null check.

Comment (hydra41) [reply CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java:259]: added null check and error code in

File: CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java [hunk 21]
```diff
@@ -325,41 +339,45 @@ private static int LoadShaders(CVKDevice cvkDevice) {
         int ret = VK_SUCCESS;
         
         try {
-            ByteBuffer vsBytes = LoadFileToDirectBuffer(CVKShaderPlaceHolder.class, "compiled/SimpleIcon.vs.spv");
-            if (vsBytes.capacity() == 0) {
-                throw new RuntimeException("Failed to load compiled/SimpleIcon.vs.spv");
+            if (vsBytes == null) {
+                vsBytes = LoadFileToDirectBuffer(CVKShaderPlaceHolder.class, "compiled/SimpleIcon.vs.spv");
```
Comment (TheTimurid) [inline CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKFPSRenderable.java:343]: These need a null check and error return.
